Lib/test/test_functools.py

```diff
@@ -746,11 +746,8 @@ def wrapper():
         self.assertEqual(wrapper.attr, 'This is a different test')
         self.assertEqual(wrapper.dict_attr, f.dict_attr)
 
-@unittest.skipUnless(c_functools, 'requires the C _functools module')
-class TestReduce(unittest.TestCase):
-    if c_functools:
-        func = c_functools.reduce
 
+class TestReduce(unittest.TestCase):
```

Comment:
This test tests only C implementation of `reduce()` if it is available. Please make both implementations be tested. See TestPartial and TestCmpToKey for example.